diff --git a/arch/arm/boot/dts/stm32mp157a-dk1.dts b/arch/arm/boot/dts/stm32mp157a-dk1.dts
index f48207dad581..f853e8947a6f 100644
--- a/arch/arm/boot/dts/stm32mp157a-dk1.dts
+++ b/arch/arm/boot/dts/stm32mp157a-dk1.dts
@@ -20,3 +20,71 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 };
+&pinctrl {
+m_can2_pins_a: m-can2-0 {
+    pins1 {
+	pinmux = <STM32_PINMUX('B', 13, AF9)>; /* CAN2_TX */
+	slew-rate = <0>;
+	drive-push-pull;
+	bias-disable;
+    };
+    pins2 {
+	pinmux = <STM32_PINMUX('B', 12, AF9)>; /* CAN2_RX */
+	bias-disable;
+    };
+};
+
+m_can2_sleep_pins_a: m_can2-sleep-0 {
+    pins {
+	pinmux = <STM32_PINMUX('B', 13, ANALOG)>, /* CAN2_TX */
+	     <STM32_PINMUX('B', 12, ANALOG)>; /* CAN2_RX */
+    };
+};
+};
+&m_can1 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&m_can1_pins_b>;
+	pinctrl-1 = <&m_can1_sleep_pins_b>;
+	status = "okay";
+};
+
+&m_can2 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&m_can2_pins_a>;
+	pinctrl-1 = <&m_can2_sleep_pins_a>;
+	status = "okay";
+};
+
+&vrefbuf {
+    regulator-min-microvolt = <2500000>;
+    regulator-max-microvolt = <2500000>;
+    vdda-supply = <&vdd>;
+    status = "okay";
+};
+
+&adc {
+    pinctrl-names = "default";
+    pinctrl-0 = <&adc12_ain_pins_a>, <&adc12_usb_cc_pins_a>;
+    vdd-supply = <&vdd>;
+    vdda-supply = <&vdd>;
+    vref-supply = <&vrefbuf>;
+    status = "okay";
+    adc1: adc@0 {
+	/*
+	 * Type-C USB_PWR_CC1 & USB_PWR_CC2 on in18 & in19.
+	 * Use at least 5 * RC time, e.g. 5 * (Rp + Rd) * C:
+	 * 5 * (56 + 47kOhms) * 5pF => 2.5us.
+	 * Use arbitrary margin here (e.g. 5us).
+	 */
+	st,min-sample-time-nsecs = <5000>;
+	/* AIN connector, USB Type-C CC1 & CC2 */
+	st,adc-channels = <0 1 6 13 18 19>;
+	status = "okay";
+    };
+    adc2: adc@100 {
+	/* AIN connector, USB Type-C CC1 & CC2 */
+	st,adc-channels = <0 1 2 6 18 19>;
+	st,min-sample-time-nsecs = <5000>;
+	status = "okay";
+    };
+};
